components/pages/ProfileTab.js

```diff
@@ -62,26 +53,46 @@ export default function ProfileTab({ navigation: { goBack }, user, refresh }) {
       .catch((err) => console.log(err));
   }
 
+  function handleSubmit(stat) {
+    const userRef = doc(db, 'users', userData.id);
+    if (userData.stats) {
+      updateDoc(userRef, { stats: arrayUnion(stat) })
+        .then(() => refresh())
```

Comment:
currently refresh doesn't do anything, but that's defined in the profile.js file. However, I noticed you're using the same refresh function for all user action handlers, which would mean the refresh function would have to refresh the entire profile page. I'm questioning if that's the right move, considering how slow our queries are right now. I think it's okay for now, but maybe something for us to consider making more selective in a future iteration.